### DISCUSSION TITLE
Support for insv files? (Insta360)

### DISCUSSION THREAD

**Discussion by DocInTheSky** [mifi/lossless-cut#1121] [Ideas]
From what I understand, insv files are just oversized mp4's with all the 360 metadata and so on. For each video you have two files (one for each lens)

Right now you can open the files and trim them, but then they are not working anymore with Insta360 software.

I assume some data was loss in the process. Also, I think it would be helpful if you could trim both video files at the same time so both are the same exact length (when I tested it I did it manually).

If i can help providing test files let me know.

**Comment by mifi** [comment]
Hi. When you say they are not working any more, do you mean that you get an error or just that the 360 projection is missing? I think what some other 360 cams are doing, like gopro, is to store both the videos as two streams in the same file.

I think opening two files as one is not something losslesscut is going to support, but you can merge two files into one file with two parallel streams using losslesscut, then after that you can cut them together. Drag drop one file, then drag the other and select streams from thsi file.

Im unfortunately not going to spend much time on trying to make losslesscut output work in a custom player like insta360 software but if it’s an obvious thing that’s missing I’m open for adding support for it. If the output also doesnt work in other players, then that is something we can look into.

**Comment by DocInTheSky** [comment]
Yeah I mean that the 360 projection is missing, it is played back as a flat video. I use the Insta360 software but there is also a plugin for Premiere Pro which I haven't tried yet but I assume that if the video goes flat on their software, it's because the 360 projection was lost.

Would you mind trying with a test video in case you figure out a workflow that works?

**Reply by mifi** [reply]
Unfortunately there are so many different 360 formats and they all do it their own way, so I think it's not going to be feasible to implement support for all the manufacturer's custom formats. If it's two separate files, then losslesscut will not be able to support that.

**Comment by jakehemmerle** [comment]
For the sake of gauging demand, this is a feature I'd also use, but I totally understand if it's not a priority relative to other things.

**Reply by mifi** [reply]
If it would be somehow possible to merge the two files into one (as multiple tracks), then it might be feasable to operate on those files, but I dunno if that's possible.

**Reply by kirill-konshin** [reply]
@mifi With the arrival of new Insta360 X4 it seems finally possible, this camera records single INSV file (just like the super old Insta350 One). Here's the link to examples: https://www.youtube.com/watch?v=K4GSfKt0Dik and Google Drive with sources https://drive.google.com/drive/folders/1YJNUOp3ggG-H9oQ0f7pm3SGI250XecOG

PS I wonder if Insta360 will ever produce a software to convert ridiculous 2-file system to 1-file system...

**Reply by mifi** [reply]
I just tried one of your files and it plays fine in losslesscut and exports successfully. actually it seems like it's just a simple MOV/MP4 but with a `.insv` file extension:
```
{
  filename: 'VID_20240414_135506_10_088.insv',
  nb_streams: 2,
  nb_programs: 0,
  format_name: 'mov,mp4,m4a,3gp,3g2,mj2',
  format_long_name: 'QuickTime / MOV',
  start_time: '0.000000',
  duration: '85.185100',
  size: '608174080',
  bit_rate: '57115535',
  probe_score: 100,
  tags: {
    major_brand: 'avc1',
    minor_version: '0',
    compatible_brands: 'avc1isom',
    creation_time: '2024-04-14T13:54:57.000000Z',
  },
}
```

I cannot confirm whether the exported file contains the metadata though. not sure how to verify that.

**Reply by kesirog** [reply]
@mifi It's not working. In order to verify that it works you have to have a insta360 camera (X3 or X4) to get insta360 studio app (for free) in there if I export the video I can see that there is no options to change directional lock or similar features that are other ways available. Also the video looks like it's recorded through peephole and it's not 360 anymore.

**Reply by mifi** [reply]
Ok, well if there's no simple way for me to verify whether the file is exported correctly, then unfortunately there's not much I can do

**Reply by kirill-konshin** [reply]
@mifi I can test it for you, let me know what should I do to merge, e.g. which command to run. I have a bunch of old INSV dual-file videos, and newer X4 single-file videos, all their apps and a camera to verify.

**Reply by mifi** [reply]
LosslessCut cannot provide support for exporting INSV unless ffmpeg supports it

**Reply by kirill-konshin** [reply]
I was under impression that it’s just an mpeg container

**Reply by mifi** [reply]
from the OP it looks like they are mp4 files with metadata

**Reply by kirill-konshin** [reply]
> from the OP it looks like they are mp4 files with metadata

Yep that's what I meant. Like I said, if you give some tips where to dig, I can help to debug.

**Reply by peterawhittaker** [reply]
I'd like to add my wish for this feature too. Insta360 has ~60% of the 360 camera market. It doesn't seem possible to losslessly edit .insv files using insta 360 Studio, and the files can be enormous. Lossless Cut support would be fantastic! Also happy to help with feedback on files if helpful.

**Reply by mifi** [reply]
maybe someone can do some experiment with exiftool after exporting from losslesscut, e.g.:

```
exiftool -tagsfromfile source.mp4 -all:all exported-file-from-losslesscut.mp4
```

and see if the file `exported-file-from-losslesscut.mp4` works in insta360 software after running that command
